content/en/docs/concepts/configuration/even-pods-spread.md

```diff
@@ -0,0 +1,270 @@
+
+## Usage
+
+### API
+
+A new `pod.spec.topologySpreadConstraints` API is introduced as below:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mypod
+spec:
+  topologySpreadConstraints:
+  - maxSkew: <integer>
+    topologyKey: <string>
+    whenUnsatisfiable: <string>
+    labelSelector: <object>
+```
+
+You can define one or multiple `topologySpreadConstraint` to instruct the kube-scheduler how to place the incoming pod with existing pods across the cluster. In particular, the fields are explained as below:
+
+- **maxSkew** describes the degree to which pods may be unevenly distributed. It's the maximum permitted difference between the number of matching pods in any two topology domains of a given topology type. It must be greater than zero.
+- **topologyKey** is the key of node labels. Nodes that have a label with this key and identical values are considered to be in the same topology. We consider each <key, value> as a "bucket", and try to put a balanced number of pods into each bucket.
+- **whenUnsatisfiable** indicates how to deal with a pod if it doesn't satisfy the spread constraint:
+    - `DoNotSchedule` (default) tells the scheduler not to schedule it.
+    - `ScheduleAnyway` tells the scheduler to still schedule it while prioritizing nodes that minimize the skew.
+- **labelSelector** is used to find matching pods. Pods that match this label selector are counted to determine the number of pods in their corresponding topology domain. See [Lable Selectors](/docs/concepts/overview/working-with-objects/labels/#label-selectors) for more details.
+
+Detailed API explanation can also be found via `kubectl explain pod.spec.topologySpreadConstraints`.
+
+### Example: One TopologySpreadConstraint
+
+Suppose we have a 4-node cluster where 3 pods labeled `foo:bar` are located in node1, node2 and node3 respectively (`P` represents Pod):
+
+```
++---------------+---------------+
+|     zoneA     |     zoneB     |
++-------+-------+-------+-------+
+| node1 | node2 | node3 | node4 |
++-------+-------+-------+-------+
+|   P   |   P   |   P   |       |
++-------+-------+-------+-------+
+```
+
+If we want an incoming pod to be evenly spread with existing pods across zones, the spec can be given as:
+
+```
+kind: Pod
+apiVersion: v1
+metadata:
+  name: mypod
+  labels:
+    foo: bar
+spec:
+  topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: zone
+    whenUnsatisfiable: DoNotSchedule
+    labelSelector:
+      matchLabels:
+        foo: bar
+...
+```
+
+`topologyKey: zone` implies the even distribution will only be applied to the nodes which have label pair "zone:<any value>" present.  And `whenUnsatisfiable: DoNotSchedule` tells the scheduler to let it stay pending if the incoming pod can’t satisfy the constraint.
+
+If we place the pod onto "zoneA", the pods distribution would become [3, 1], hence the actual skew is 2 (3 - 1) which violates `maxSkew: 1`, so in this example, the incoming pod can only be placed onto "zoneB":
```

Comment:
```suggestion
If the control plane places the incoming Pod onto "zoneA", the pods distribution would become [3, 1], hence the actual skew is 2 (3 - 1) which violates `maxSkew: 1`. In this example, the incoming pod can only be placed onto "zoneB":
```

Reply:
We can just say:

> If the Pod is placed into "zoneA"

BTW: "onto", or "into"? Any suggestion?